### PR TITLE
 Make service launch type configurable 

### PR DESCRIPTION
**Pull request by lassebenni**

### CODE REVIEW

**Comment by lassebenni** [comment]
@fatbasstard Would you mind taking a look at this PR? We want to use it to provision an EC2 instance in ECS.

File: variables.tf [hunk 1]
```diff
@@ -130,6 +130,12 @@ variable "secrets" {
   description = "Map containing secrets to expose to the docker container"
 }
 
+variable "service_launch_type" {
+  type        = string
+  default     = "FARGATE"
+  description = "The service launch type: either FARGATE or EC2"
+}
```
Comment (fatbasstard) [inline variables.tf:137]: Can you add a validation check on this? So we can ensure it's always one of these 2 values?

See: https://www.terraform.io/docs/language/values/variables.html#custom-validation-rules

Something like this:
``` 
 validation {
    condition     = contains(["FARGATE", "EC2", "eur", "nsa"], var. service_launch_type)
    error_message = "Allowed values for service_launch_type are \"FARGATE\",  or \"EC2\"."
  }

**Comment by fatbasstard** [comment]
Looks good 👍  I'm just not sure if there are any conditionals in the resource (Like certain attributes only apply to `FARGATE` for insance). Have you tested that?

**Comment by lassebenni** [comment]
> Looks good 👍 I'm just not sure if there are any conditionals in the resource (Like certain attributes only apply to `FARGATE` for insance). Have you tested that?

Good point, I added the only one I could find:

```
assign_public_ip = var.service_launch_type == "FARGATE" ? var.public_ip : false
```

**Review by fatbasstard — APPROVED** [review]

**Pull request merged**